src/component/vocabulary/VocabularyNameBadgeButton.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState } from "react";
+import { AssetData } from "../../model/Asset";
+import VocabularyNameBadge from "./VocabularyNameBadge";
+import { UncontrolledTooltip } from "reactstrap";
+import { Routing } from "../../util/Routing";
+import { isLoggedIn } from "../../util/Authorization";
+import Routes from "../../util/Routes";
+import { useDispatch, useSelector } from "react-redux";
+import TermItState from "../../model/TermItState";
+import VocabularyUtils from "../../util/VocabularyUtils";
+import { Link } from "react-router-dom";
+import { ThunkDispatch } from "../../util/Types";
+import { getLabel } from "../../action/AsyncActions";
+import Utils from "../../util/Utils";
+
+interface VocabularyNameBadgeButtonProps {
+  vocabulary?: AssetData;
+  className?: string;
+  termIri: string;
+  section: string;
+}
+
+const VocabularyNameBadgeButton: React.FC<VocabularyNameBadgeButtonProps> = ({
+  className,
+  vocabulary,
+  termIri,
+  section,
+}) => {
+  const dispatch: ThunkDispatch = useDispatch();
+  const user = useSelector((state: TermItState) => state.user);
+  const [label, setLabel] = useState<string>();
+  // Unique identifier to associate tooltip with correct term on the page (same term can appear multiple times in diff sections
+  const id = `vocabulary-name-badge-${section}` + Utils.hashCode(termIri);
+
+  // We need to know the label in advance. If the tooltip's content changes, the arrow gets misplaced
+  React.useEffect(() => {
+    if (!vocabulary) return;
+    dispatch(getLabel(vocabulary!.iri!)).then((data) => {
+      if (data) {
+        setLabel(data);
+      } else {
+        setLabel(vocabulary!.iri!);
+      }
+    });
+  }, [dispatch, vocabulary]);
+
+  if (!vocabulary) {
+    return null;
+  }
+
+  const iri = VocabularyUtils.create(vocabulary!.iri!);
+  const path = Routing.getTransitionPath(
+    isLoggedIn(user)
+      ? Routes.vocabularySummary
+      : Routes.publicVocabularySummary,
+    {
+      params: new Map([["name", iri.fragment]]),
+      query: new Map([["namespace", iri.namespace!]]),
+    }
+  );
```

Comment:
Could use `Vocabularies.getVocabularyRoutingOptions` from Routing.

Reply:
I looked into the method and it only accepts `Vocabulary` as a parameter. The components which render the badge have the vocabulary only as an `AssetData`. Is there a way how to convert it? I couldn't find any... The code is not that long, so I don't think it is doing that much harm, but I get your point. 
So do we leave it? Or should I fetch the vocabularies? Or is there a way how to sufficiently parse `AssetData` into `Vocabulary`? @ledsoft

Reply:
Ok, sorry, I saw the variable `vocabulary` and didn't bother checking its type. Let's leave it as is.